frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/ColumnPicker/ColumnPicker.tsx

```diff
@@ -0,0 +1,153 @@
+import classNames from "classnames";
+import type { FocusEvent, MouseEvent, KeyboardEvent, ReactNode } from "react";
+import { useRef, useState, useMemo, useEffect } from "react";
+import { t } from "ttag";
+
+import { QueryColumnPicker } from "metabase/common/components/QueryColumnPicker";
+import useSequencedContentCloseHandler from "metabase/hooks/use-sequenced-content-close-handler";
+import { color } from "metabase/lib/colors";
+import { Button, Icon, Input, Popover, FocusTrap } from "metabase/ui";
+import { getThemeOverrides } from "metabase/ui/theme";
+import * as Lib from "metabase-lib";
+
+import styles from "./ColumnPicker.module.css";
+
+type ColumnInputProps = {
+  query: Lib.Query;
+  stageIndex: number;
+  columns: Lib.ColumnMetadata[];
+  label?: string;
+  value: Lib.ColumnMetadata;
+  onChange: (column: Lib.ColumnMetadata) => void;
+};
+
+const theme = getThemeOverrides();
+
+export function ColumnPicker({
+  query,
+  stageIndex,
+  columns,
+  label,
+  value,
+  onChange,
+}: ColumnInputProps) {
+  const columnGroups = useMemo(() => Lib.groupColumns(columns), [columns]);
+
+  const [open, setOpen] = useState(false);
+  const button = useRef<HTMLButtonElement>(null);
+
+  function handleOpen() {
+    setOpen(true);
+  }
+
+  function handleClose() {
+    setOpen(false);
+    button.current?.focus();
+  }
+
+  function handleBlur(evt: FocusEvent<HTMLDivElement>) {
+    if (!evt.currentTarget || !evt.relatedTarget) {
+      return;
+    }
+    if (!evt.currentTarget.contains(evt.relatedTarget as Node)) {
+      setTimeout(() => setOpen(false), 100);
```

Comment:
This will cause issues with onmounts before timeout. Please use a hook (maybe from react-use) that handles that.